dnsprovider/aws.go

```diff
@@ -46,6 +47,26 @@ type AWSProvider struct {
 	DryRun bool
 }
 
+// NewAWSProvider initializes a new AWS Route53 based DNSProvider.
+func NewAWSProvider(dryRun bool) (DNSProvider, error) {
+	config := aws.NewConfig()
+
+	session, err := session.NewSessionWithOptions(session.Options{
```

Comment:
> If [...] SharedConfigEnable option is used to create the Session the full shared config values will be loaded. This includes credentials, region, and support for assume role. In addition the Session will load its configuration from both the shared config file (\~/.aws/config) and shared credentials file (\~/.aws/credentials). Both files have the same format.

I think this will make it easier to use locally at least.